drivers/i3c/i3c_npcx.c

```diff
@@ -2144,41 +2884,55 @@ static int npcx_i3c_init(const struct device *dev)
 		return ret;
 	}
 
+	/* Lock initial */
 	k_mutex_init(&data->lock_mutex);
 	k_sem_init(&data->sync_sem, 0, 1);
 	k_sem_init(&data->ibi_lock_sem, 1, 1);
+	k_sem_init(&data->target_lock_sem, 1, 1);
+	k_sem_init(&data->target_event_lock_sem, 1, 1);
 
 	ret = i3c_addr_slots_init(dev);
 	if (ret != 0) {
 		LOG_ERR("Addr slots init fail %d", ret);
 		return ret;
 	}
 
-	ctrl_config->is_secondary = false; /* Currently can only act as primary controller. */
-	ctrl_config->supported_hdr = I3C_MSG_HDR_DDR; /* HDR-DDR mode is supported. */
-	ctrl_config->scl.i3c = config->clocks.i3c_pp_scl_hz; /* Set I3C frequency */
+	/* Set controller default configuration */
+	config_cntlr->is_secondary = false; /* Currently can only act as primary controller. */
```

Comment:
it makes more sense to have that dts prop to only define if it is a secondary controller. If it's a secondary controller, then it should initialize as a target, if it is a primary controller, then it should initialize as a controller.

The only way to have to 'switch' at runtime would be to do a controller handoff rather than calling the configure function to have it initialize as a target.

Reply:
Done